File: src/algebra1_exponent_rules_medium.py
"""
algebra1 - exponent_rules (medium)
Generated: 2026-02-22T03:58:00.226289
"""

from problem_utils import *

def generate():
    problem_type = randint(1, 6)

    if problem_type == 1:
        # x^a * x^b with larger exponents
        base_var = choice([x, y, a, b])
        exp1 = randint(2, 10)
        exp2 = randint(2, 10)
        ans = base_var**(exp1 + exp2)

        return problem(
            question=f"Simplify: ${latex(base_var)}^{{{exp1}}} \\cdot {latex(base_var)}^{{{exp2}}}$",
            answer=ans,
            difficulty=(1300, 1400),
            topic="algebra1/exponent_rules",
            solution=steps(
                f"Product rule: $x^a \\cdot x^b = x^{{a+b}}$",
                f"$= {latex(base_var)}^{{{exp1}+{exp2}}} = {latex(ans)}$"
            )
        )

    elif problem_type == 2:
        # (x^a)^b with varied exponents
        base_var = choice([x, y, a, b])
        exp1 = randint(2, 7)
        exp2 = randint(2, 5)
        ans = base_var**(exp1 * exp2)

        return problem(
            question=f"Simplify: $\\left({latex(base_var)}^{{{exp1}}}\\right)^{{{exp2}}}$",
            answer=ans,
            difficulty=(1300, 1400),
            topic="algebra1/exponent_rules",
            solution=steps(
                f"Power rule: $(x^a)^b = x^{{ab}}$",
                f"$= {latex(base_var)}^{{{exp1} \\cdot {exp2}}} = {latex(ans)}$"
            )
        )

    elif problem_type == 3:
        # Quotient rule with large exponents
        base_var = choice([x, y, a, b])
        exp1 = randint(5, 12)
        exp2 = randint(1, 4)
        ans = base_var**(exp1 - exp2)

        return problem(
            question=f"Simplify: $\\frac{{{latex(base_var)}^{{{exp1}}}}}{{{latex(base_var)}^{{{exp2}}}}}$",
            answer=ans,
            difficulty=(1300, 1400),
            topic="algebra1/exponent_rules",
            solution=steps(
                f"Quotient rule: $x^a / x^b = x^{{a-b}}$",
                f"$= {latex(base_var)}^{{{exp1}-{exp2}}} = {latex(ans)}$"
            )
        )

    elif problem_type == 4:
        # Negative exponent
        base_var = choice([x, y, a, b])
        exp1 = randint(1, 7)
        ans = 1 / base_var**exp1

        return problem(
            question=f"Simplify: ${latex(base_var)}^{{-{exp1}}}$",
            answer=ans,
            difficulty=(1400, 1500),
            topic="algebra1/exponent_rules",
            solution=steps(
                f"Negative exponent rule: $x^{{-a}} = \\frac{{1}}{{x^a}}$",
                f"${latex(base_var)}^{{-{exp1}}} = \\frac{{1}}{{{latex(base_var)}^{{{exp1}}}}}$"
            )
        )

    elif problem_type == 5:
        # Combined: (x^a * x^b)^c with coefficient
        base_var = choice([x, y, a, b])
        exp1 = randint(2, 5)
        exp2 = randint(2, 5)
        exp3 = randint(2, 4)
        ans = base_var**((exp1+exp2)*exp3)

        return problem(
            question=f"Simplify: $\\left({latex(base_var)}^{{{exp1}}} \\cdot {latex(base_var)}^{{{exp2}}}\\right)^{{{exp3}}}$",
            answer=ans,
            difficulty=(1500, 1600),
            topic="algebra1/exponent_rules",
            solution=steps(
                f"First: $x^{{{exp1}}} \\cdot x^{{{exp2}}} = x^{{{exp1+exp2}}}$ (product rule)",
                f"Then: $(x^{{{exp1+exp2}}})^{{{exp3}}} = x^{{{exp1+exp2} \\cdot {exp3}}} = {latex(ans)}$ (power rule)"
            )
        )

    else:
        # Simplify expression with coefficients and negative exponents
        base_var = choice([x, y])
        c1 = randint(2, 8)
        c2 = randint(2, 8)
        exp1 = randint(2, 6)
        exp2 = randint(exp1+1, exp1+5)
        # c1*x^exp1 / (c2*x^exp2)
        final_coeff = Rational(c1, c2)
        final_exp = exp1 - exp2

        if final_coeff == 1:
            ans = base_var**final_exp
        else:
            ans = final_coeff * base_var**final_exp

        return problem(
            question=f"Simplify: $\\frac{{{c1}{latex(base_var)}^{{{exp1}}}}}{{{c2}{latex(base_var)}^{{{exp2}}}}}$",
            answer=ans,
            difficulty=(1450, 1600),
            topic="algebra1/exponent_rules",
            solution=steps(
                f"Separate coefficient and variable: $\\frac{{{c1}}}{{{c2}}} \\cdot \\frac{{{latex(base_var)}^{{{exp1}}}}}{{{latex(base_var)}^{{{exp2}}}}}$",
                f"$= {latex(final_coeff)} \\cdot {latex(base_var)}^{{{exp1}-{exp2}}}$",
                f"$= {latex(ans)}$"
            )
        )

emit(generate())

File: src/algebra1_exponent_rules_medium.jl
# algebra1 - exponent_rules (medium)
# Generated: 2026-03-08T19:58:37.321212

include(joinpath(@__DIR__, "..", "julia", "src", "ProblemUtils.jl"))
using .ProblemUtils

@script x begin
    set_topic!("algebra1/exponent_rules")
    
    problem_type = rand(1:6)
    
    if problem_type == 1
        # Product rule: x^a * x^b = x^(a+b)
        a = randint(2, 8)
        b = randint(2, 8)
        ans = a + b
        
        problem(
            question="Simplify: \$x^{$a} \\cdot x^{$b}\$",
            answer="x^$ans",
            difficulty=(1200, 1400),
            solution=steps(
                "Apply the product rule: \$x^a \\cdot x^b = x^{a+b}\$",
                sol("Calculation", "x^{$a} \\cdot x^{$b} = x^{$a + $b}"),
                sol("Answer", "x^{$ans}")
            ),
            time=60
        )
        
    elseif problem_type == 2
        # Quotient rule: x^a / x^b = x^(a-b)
        a = randint(5, 12)
        b = randint(2, 6)
        ans = a - b
        
        problem(
            question="Simplify: \$\\frac{x^{$a}}{x^{$b}}\$",
            answer="x^$ans",
            difficulty=(1200, 1400),
            solution=steps(
                "Apply the quotient rule: \$\\frac{x^a}{x^b} = x^{a-b}\$",
                sol("Calculation", "\\frac{x^{$a}}{x^{$b}} = x^{$a - $b}"),
                sol("Answer", "x^{$ans}")
            ),
            time=60
        )
        
    elseif problem_type == 3
        # Power rule: (x^a)^b = x^(ab)
        a = randint(2, 6)
        b = randint(2, 5)
        ans = a * b
        
        problem(
            question="Simplify: \$(x^{$a})^{$b}\$",
            answer="x^$ans",
            difficulty=(1200, 1400),
            solution=steps(
                "Apply the power rule: \$(x^a)^b = x^{ab}\$",
                sol("Calculation", "(x^{$a})^{$b} = x^{$a \\cdot $b}"),
                sol("Answer", "x^{$ans}")
            ),
            time=60
        )
        
    elseif problem_type == 4
        # Combined product and power: (x^a)^b * x^c
        a = randint(2, 5)
        b = randint(2, 4)
        c = randint(2, 6)
        ans = a * b + c
        
        problem(
            question="Simplify: \$(x^{$a})^{$b} \\cdot x^{$c}\$",
            answer="x^$ans",
            difficulty=(1400, 1600),
            solution=steps(
                "Apply the power rule: \$(x^{$a})^{$b} = x^{$(a*b)}\$",
                "Apply the product rule: \$x^{$(a*b)} \\cdot x^{$c} = x^{$(a*b) + $c}\$",
                sol("Answer", "x^{$ans}")
            ),
            time=90
        )
        
    elseif problem_type == 5
        # Combined quotient and power: (x^a)^b / x^c
        a = randint(3, 6)
        b = randint(2, 4)
        c = randint(2, 8)
        ans = a * b - c
        
        problem(
            question="Simplify: \$\\frac{(x^{$a})^{$b}}{x^{$c}}\$",
            answer="x^$ans",
            difficulty=(1400, 1600),
            solution=steps(
                "Apply the power rule: \$(x^{$a})^{$b} = x^{$(a*b)}\$",
                "Apply the quotient rule: \$\\frac{x^{$(a*b)}}{x^{$c}} = x^{$(a*b) - $c}\$",
                sol("Answer", "x^{$ans}")
            ),
            time=90
        )
        
    else
        # Multi-step: (x^a * x^b)^c / x^d
        a = randint(2, 4)
        b = randint(2, 4)
        c = randint(2, 3)
        d = randint(3, 8)
        ans = (a + b) * c - d
        
        problem(
            question="Simplify: \$\\frac{(x^{$a} \\cdot x^{$b})^{$c}}{x^{$d}}\$",
            answer="x^$ans",
            difficulty=(1500, 1800),
            solution=steps(
                "Apply the product rule inside parentheses: \$x^{$a} \\cdot x^{$b} = x^{$(a+b)}\$",
                "Apply the power rule: \$(x^{$(a+b)})^{$c} = x^{$((a+b)*c)}\$",
                "Apply the quotient rule: \$\\frac{x^{$((a+b)*c)}}{x^{$d}} = x^{$((a+b)*c) - $d}\$",
                sol("Answer", "x^{$ans}")
            ),
            time=120
        )
    end
end